client/lib/explat/internals/anon-id.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * Internal dependencies
+ */
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { tracksAnonymousUserId } from 'calypso/lib/analytics/ad-tracking';
+import userUtils from 'calypso/lib/user/utils';
+import { logError, isDevelopmentMode } from './log-error';
+
+// Using typescript to ensure we are being safe in SSR
+declare const window: undefined | ( Window & typeof globalThis );
+
+const logErrorOrThrowInDevelopmentMode = ( message: string ) => {
+	if ( isDevelopmentMode ) {
+		throw new Error( message );
+	} else {
+		logError( { message } );
+	}
+};
+
+/**
+ * setInterval, but it runs first callback immediately instead of after interval.
+ *
+ * @param f The callback function
+ * @param intervalMs The interval in milliseconds
+ */
+const immediateStartSetInterval = ( f: () => void, intervalMs: number ) => {
+	f();
+	return setInterval( f, intervalMs );
+};
+
+let initializeAnonIdPromise: null | Promise< string | null > = null;
+const anonIdPollingIntervalMilliseconds = 50;
+const anonIdPollingIntervalMaxAttempts = 100; // 50 * 100 = 5000 = 5 seconds
```

Comment:
Is this timing ok?

Reply:
As discussed on the P2 thread, I think it's OK, but it's worth adding an explanation here as an inline comment because five seconds sounds like a long time in general.